.github/workflows/e2e-functional.yml

```diff
@@ -81,50 +86,48 @@ jobs:
       nightly: ${{ inputs.nightly }}
     secrets: inherit
 
-  update-final-status-linux:
+  update-final-status:
+    name: Update final status
     runs-on: ubuntu-22.04
     needs:
+      - prepare-matrix
       - e2e-tests
+    if: always()
     steps:
-      - uses: mattermost/actions/delivery/update-commit-status@main
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
+      - name: Update final status for all platforms
+        uses: actions/github-script@ed597411d8f924073f98dfc5c65a23a2325f34cd # v8.0.0
         with:
-          repository_full_name: ${{ github.repository }}
-          commit_sha: ${{ github.event.pull_request.head.sha || github.sha }}
-          context: e2e/linux
-          description: Completed with ${{ needs.e2e-tests.outputs.NEW_FAILURES_LINUX }} failures
-          status: ${{ needs.e2e-tests.outputs.STATUS_LINUX }}
-          target_url: ${{ needs.e2e-tests.outputs.REPORT_LINK_LINUX }}
+          github-token: ${{ github.token }}
+          script: |
```

Comment:
Nit: do these scripts need to be in-line? Or can we store them in a different file? Seems odd to mix YAML and JS.